### PR TITLE
Add alar2 to repair-script-library

### PR DESCRIPTION
**Pull request by malachma**
This is new version of ALAR (alar2) it is intended to replace the current bash script based ALAR. 
Any new development and bug-fixing will be performed on alar2 only. 
There will be another PR again "Azure/azure-cli-extensions" for additional functionality as custom.py and repair_utils.py had to be extended/altered

### CODE REVIEW

**Comment by swbae31** [comment]
I see some a lot of non-code files, can we remove the release files?

**Comment by swbae31** [comment]
Let's revive ALAR 1 for existing users.

**Comment by swbae31** [comment]
Let's change the map.json file description so that we can tell users that ALAR 2 might take some time.

File: map.json [hunk 1]
```diff
@@ -34,10 +34,15 @@
...
-	{
-		"id" : "linux-alar-fki",
-		"path" : "src/linux/linux-alar-fki.sh",
-		"description" : "alar-fki allows to recover a failed VM. Three recover scenarios are possible: fstab, initrd and kernel. NOTE: use option --run-on-repair. Se the docu for more details: https://github.com/Azure/repair-script-library/blob/master/src/linux/common/helpers/alar/README.md"
+    {
+        "id" : "linux-alar-fki",
+        "path" : "src/linux/linux-alar-fki.sh",
+        "description" : "DEPRECATED: It is recommended to use linux-alar2 instead. alar-fki allows to recover a failed VM. Three recover scenarios are possible: fstab, initrd and kernel. NOTE: use option --run-on-repair. Se the docu for more details: https://github.com/Azure/repair-script-library/blob/master/src/linux/common/helpers/alar/README.md"
+    },
+	{
```
Comment (swbae31) [inline map.json:42]: The spacing seems uneven here.

**Comment by malachma** [comment]
Hi Sungwoo, anything else I should improve or verify?

File: map.json (hunk 1 above)
Comment (malachma) [reply map.json:42]: Spacing I have corrected

File: src/linux/linux-alar2.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+. ./src/linux/common/setup/init.sh
+
+Log-Output "Starting the recovery"
+cd ./src/linux/common/helpers/alar2
+/root/.cargo/bin/cargo build -q --release
```
[the remark this answers is not in the document]
Comment (malachma) [reply src/linux/linux-alar2.sh:6]: Yes, this "/root/.cargo/bin/cargo build -q --release"  is the path to the the installation directory  of rust. Cargo is used to build the application.

File: map.json [hunk 3]
```diff
@@ -34,10 +34,15 @@
...
+        {
+        	"id" : "linux-alar-fki",
+        	"path" : "src/linux/linux-alar-fki.sh",
+        	"description" : "DEPRECATED: It is recommended to use linux-alar2 instead. alar-fki allows to recover a failed VM. Three recover scenarios are possible: fstab, initrd and kernel. NOTE: use option --run-on-repair. See the docu for more details: https://github.com/Azure/repair-script-library/blob/master/src/linux/common/helpers/alar/README.md"
+    	},
+	{
+		"id" : "linux-alar2",
+		"path" : "src/linux/linux-alar2.sh",
+		"description" : "alar2 allows to recover a failed VM. Various actions are available like: fstab, initrd and kernel. NOTE: use option --run-on-repair. See the docu for more details: https://github.com/Azure/repair-script-library/blob/master/src/linux/common/helpers/alar2/README.md. Please be aware that alar2 needs to be build on the recovery VM first. This takes about 2min. Please be patient."
```
Comment (swbae31) [inline map.json:45]: 1) Remove "Please be patient." 
2) Please run alar2 on a repair vm with the az vm repair run --run-on-repair command.

**Comment by swbae31** [comment]
Let's add a Readme for ALAR2 in the similar fashion as ALAR1 readme: https://github.com/Azure/repair-script-library/blob/alar2-test/src/linux/common/helpers/alar/README.md

**Comment by malachma** [comment]
This pull request is not writable for me anymore.
We need at first to merge it into main. Then I can add the missing files

**Comment by malachma** [comment]
This PR can be closed. It got fixed by another PR

**Pull request closed without merging**